packages/next/src/server-build.ts

```diff
@@ -2089,6 +2089,25 @@ export async function serverBuild({
       // with that routing section
       ...afterFilesRewrites,
 
+      // Ensure that after we normalize `afterFilesRewrites`, unmatched actions are routed to the correct handler
+      // e.g. /foo/.action -> /foo.action. This should only ever match in cases where we're routing to an action handler
+      // and the rewrite normalization led to something like /foo/$1$rscsuff, and $1 had no match.
+      // This is meant to have parity with the .rsc handling below.
+      ...(hasActionOutputSupport
+        ? [
+            {
+              src: `${path.posix.join('/', entryDirectory, '/\\.action$')}`,
+              dest: `${path.posix.join('/', entryDirectory, '/index.action')}`,
+              check: true,
+            },
+            {
+              src: `${path.posix.join('/', entryDirectory, '(.+)/\\.action$')}`,
```

Comment:
```suggestion
              src: `${path.posix.join('/', entryDirectory, '(.+)/\\.action$')}`,
```
Same as this thread in case that one should be updated https://github.com/vercel/vercel/pull/11688/files#r1625164267